src/main/java/org/zalando/nakadi/service/EventStreamWriterBinary.java

```diff
@@ -44,11 +63,59 @@ public int writeBatch(final OutputStream os, final Cursor cursor, final List<Str
         final byte[] offset = cursor.getOffset().getBytes(StandardCharsets.UTF_8);
         os.write(offset);
         byteCount += offset.length;
+
+        os.write(B_CURSOR_PARTITION_END);
+        if (!events.isEmpty()) {
+            os.write(B_EVENTS_ARRAY_BEGIN);
+            for (int i = 0; i < events.size(); i++) {
+                final byte[] event = events.get(i);
+                os.write(event);
+                byteCount += event.length;
+                if (i < (events.size() - 1)) {
+                    os.write(B_COMMA_DELIM);
+                } else {
+                    os.write(B_CLOSE_BRACKET);
+                }
+            }
+        }
+        os.write(B_CLOSE_CURLY_BRACKET);
+        os.write(B_BATCH_SEPARATOR);
+
+        os.flush();
+
+        return byteCount;
+    }
+
+    @Override
+    public int writeSubscriptionBatch(final OutputStream os, final SubscriptionCursor cursor,
+                                      final List<ConsumedEvent> events,
+                                      final Optional<String> metadata) throws IOException {
+        int byteCount = B_FIXED_BYTE_COUNT_SUBSCRIPTION;
+
+        os.write(B_CURSOR_PARTITION_BEGIN);
+        final byte[] partition = cursor.getPartition().getBytes(StandardCharsets.UTF_8);
```

Comment:
@rcillo There is `import static com.google.common.base.Charsets.UTF_8` in this file